src/Microsoft.CodeQuality.Analyzers/Core/Maintainability/UseNameofInPlaceOfString.cs

```diff
@@ -0,0 +1,152 @@
+                                                                         isEnabledByDefault: true,
+                                                                         description: s_localizableDescription,
+                                                                         helpLinkUri: "https://github.com/dotnet/roslyn-analyzers/blob/master/src/Microsoft.CodeQuality.Analyzers/Microsoft.CodeQuality.Analyzers.md#maintainability",
+                                                                         customTags: WellKnownDiagnosticTags.Telemetry);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(RuleWithSuggestion);
+
+        public override void Initialize(AnalysisContext analysisContext)
+        {
+            analysisContext.EnableConcurrentExecution();
+            analysisContext.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            analysisContext.RegisterOperationAction(AnalyzeArgument, OperationKind.Argument);
+        }
+
+        private void AnalyzeArgument(OperationAnalysisContext context)
+        {
+            var argument = (IArgumentOperation)context.Operation;
+            if ((argument.Value.Kind != OperationKind.Literal 
+                || argument.Value.Type.SpecialType != SpecialType.System_String))
+            {
+                return;
+            }
+
+            if (argument.Parameter == null)
+            {
+                return;
+            }
+
+            var stringText = (string)argument.Value.ConstantValue.Value;
+            var properties = ImmutableDictionary<string, string>.Empty.Add(StringText, stringText);
+
+            var matchingParameter = argument.Parameter;
+
+            switch (matchingParameter.Name)
+            {
+                case ParamName:
+                    var parametersInScope = GetParametersInScope(context);
+                    if (HasAMatchInScope(stringText, parametersInScope))
+                    {
+                        context.ReportDiagnostic(Diagnostic.Create(
+                            RuleWithSuggestion, argument.Value.Syntax.GetLocation(), properties, stringText ));
+                    }
+                    return;
+                case PropertyName:
+                    var propertiesInScope = GetPropertiesInScope(context);
+                    if (HasAMatchInScope(stringText, propertiesInScope))
+                    {
+                        context.ReportDiagnostic(Diagnostic.Create(
+                            RuleWithSuggestion, argument.Value.Syntax.GetLocation(), properties, stringText));
+                    }
+                    return;
+                default:
+                    return;
+            }
+        }
+
+        private IEnumerable<string> GetPropertiesInScope(OperationAnalysisContext context)
+        {
+            var containingType = context.ContainingSymbol.ContainingType;
```

Comment:
is there we do this rather than using Lookup symbol?

Reply:
ignore me. I saw where it is intended to be used and why it only care about parameter and property.